### PR TITLE
autogen: add libtool patch for NVIDIA HPC compilers

### PR DESCRIPTION
**Pull request by yfguo**
## Pull Request Description

NVIDIA HPC compilers add nvc, nvc++, nvfortran in addition to the PGI
compiler binary names. This patch fixes libtool for them.

This should fixes the previous attempt which broke the PGI build.

## Author Checklist
* [ ] Reference appropriate issues (with "Fixes" or "See" as appropriate)
* [ ] Remove xfail from the test suite when fixing a test
* [ ] Commits are self-contained and do not do two things at once
* [ ] Commit message is of the form: `module: short description` and follows [good practice](https://chris.beams.io/posts/git-commit/)
* [ ] Passes whitespace checkers
* [ ] Passes warning tests
* [ ] Passes all tests
* [ ] Add comments such that someone without knowledge of the code could understand
* [ ] You or your company has a signed contributor's agreement on file with Argonne
* [ ] For non-Argonne authors, request an explicit comment from your companies PR approval manager

### CODE REVIEW

**Comment by yfguo** [comment]
test:mpich/custom
compiler: pgi
netmod: ch4:ofi

**Comment by yfguo** [comment]
The building problem with PGI introduced by the last attempt of this libtool patch is fixed. Need to look at the alignment failure.

**Comment by yfguo** [comment]
test:mpich/ch4/ofi

Just to test for sanity.

**Comment by yfguo** [comment]
test:mpich/ch4/ofi
test:mpich/custom
compiler: pgi
netmod: ch4:ofi

**Comment by yfguo** [comment]
Actually, I need to try static build as well to make sure this works as expected.

**Comment by yfguo** [comment]
test:mpich/ch4/ofi
test:mpich/custom
compiler: pgi
netmod: ch4:ofi


The PR is ready for review again. Tested locally for both static and dynamic build.

File: maint/patches/optional/confdb/nv-compiler.patch [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
++	*Portland\ Group* | *NVIDIA\ Compilers* | *PGI\ Compilers*)
+ 	  _LT_TAGVAR(lt_prog_compiler_wl, $1)='-Wl,'
+ 	  _LT_TAGVAR(lt_prog_compiler_pic, $1)='-fpic'
+ 	  _LT_TAGVAR(lt_prog_compiler_static, $1)='-Bstatic'
+@@ -5262,12 +5261,12 @@
+ 	tmp_addflag=' $pic_flag'
+ 	tmp_sharedflag='-shared'
+ 	case $cc_basename,$host_cpu in
+-        pgcc*)				# Portland Group C compiler
++        pgcc* | nvc )				# NVIDIA HPC C++ compiler
```
Comment (raffenet) [inline maint/patches/optional/confdb/nv-compiler.patch:40]: comment should be `NVIDIA HPC C compiler`?

Comment (yfguo) [reply maint/patches/optional/confdb/nv-compiler.patch:40]: True. Let me fix it.

**Comment by yfguo** [comment]
test:mpich/ch4/ofi
test:mpich/custom
compiler: pgi
netmod: ch4:ofi

**Review by raffenet — APPROVED** [review]

**Comment by yfguo** [comment]
test:mpich/ch4/ofi
test:mpich/custom
compiler: pgi
netmod: ch4:ofi

**Pull request merged**